src/Bicep.Core/LanguageConstants.cs

```diff
@@ -272,7 +273,23 @@ public static IEnumerable<TypeProperty> CreateResourceProperties(ResourceTypeRef
             {
                 yield return prop;
             }
+            yield return new TypeProperty("location", String);
+
+            yield return new TypeProperty("tags", Tags);
+
+            yield return new TypeProperty("properties", Object);
+
+            foreach (var prop in KnownTopLevelResourceProperties())
+            {
+                yield return prop;
+            }
+
+            var resourceRefArray = new TypedArrayType(ResourceRef, TypeSymbolValidationFlags.Default);
+            yield return new TypeProperty(ResourceDependsOnPropertyName, resourceRefArray, TypePropertyFlags.WriteOnly | TypePropertyFlags.DisallowAny);
```

Comment:
FYI, I think it should be possible to remove dependsOn here as it should be replaced here: https://github.com/Azure/bicep/blob/f1a4461a84bdb7b3c0f5e329b64387b0ba73fe72/src/Bicep.Core/TypeSystem/Az/AzResourceTypeProvider.cs#L179

Reply:
dependsOn, location, etc. was here before. I just split to the "built-in" ones and the "known" ones.

Reply:
Just to clarify - I'm only talking about dependsOn. DependsOn a special case because it's not a 'real' API property. It's no longer needed here as it's injected by the type provider (see code I linked)

Reply:
ok, in that case - I remove it.

Reply:
I'm happy to make this change in a separate PR btw - it's not necessary for this change. I figured it would simplify things as it removes the need to separate out `CreateResourceProperties` and `KnownTopLevelResourceProperties` into two methods.

Reply:
I removed the dependsOn, however others and the split I'd rather not touch - it seems to me like there is some logic in the AzResouceTypePRovider behind it that I don't fully get all the implications.